src/Utilities/Compiler/Extensions/IMethodSymbolExtensions.cs

```diff
@@ -709,9 +709,7 @@ public static bool IsAutoPropertyAccessor(this IMethodSymbol methodSymbol)
         /// Check if the given <paramref name="methodSymbol"/> is an implicitly generated method for top level statements.
         /// </summary>
         public static bool IsTopLevelStatementsEntryPointMethod([NotNullWhen(true)] this IMethodSymbol? methodSymbol)
-            => methodSymbol?.ContainingType.IsTopLevelStatementsEntryPointType() == true &&
-               methodSymbol.IsStatic &&
```

Comment:
I think we should still retain the `IsStatic` check.

Reply:
@mavasani It looks redundant to me. If the name is `<Main>$`, then the compiler must have emitted it as static. Any particular reason to keep it?

(As it's not a big deal - I'm going to add it now, but curious about what difference does it make)